src/spline.c

```diff
@@ -93,14 +93,37 @@ static void _twin_spline_decompose(twin_path_t *path,
     _twin_path_sdraw(path, spline->a.x, spline->a.y);
 
     while (!is_flat(spline, tolerance_squared)) {
-        int shift = 1;
+        twin_dfixed_t hi = TWIN_SFIXED_ONE * TWIN_SFIXED_ONE, lo = 0;
+        twin_dfixed_t best = 0, t;
+        twin_dfixed_t best_error = 0, error;
         twin_spline_t left, right;
 
-        /* FIXME: Find the optimal shift value to decompose the spline */
-        do {
-            _de_casteljau(spline, shift, &left, &right);
-            shift++;
-        } while (!is_flat(&left, tolerance_squared));
+        while (true) {
+            t = (hi + lo) >> 1;
+            if (best == t)
+                break;
+
+            _de_casteljau(spline, t, &left, &right);
+
+            error = _twin_spline_error_squared(&left);
+            if (error < best_error)
+                break;
+
+            /* Left is close enough to the bezier curve */
+            if (error <= tolerance_squared) {
+                best_error = error;
```

Comment:
I suggest renaming `best_error` to either `farthest_distance` or `optimal_distance`, as your intent isn't clearly reflected in the code.

Based on the current variable name, the purpose of `best_error` isn't immediately obvious.

Initially, I thought it represented a lower bound, but after reviewing your development process, I realized its actual purpose is to find the farthest distance from point a to point d.

Also, rename `best` to `best_t` or `optimal_t`. 
Avoid using only adjectives as variable names. Make sure what the variable it is.

Refs:
1. https://hackmd.io/UIOMextwQbqIcZaKuFkl4Q#iterative-splines

Reply:
> Also, rename best to `best_t` or `optimal_t`.

The suffix `_t` refers to the derived type using `typedef`. In this case, we can say, `t_optimal`.